r/tests/testthat/test-dplyr-mutate.R

```diff
@@ -185,6 +164,238 @@ test_that("mutate with single value for recycling", {
+  expect_dplyr_equal(input %>% mutate(y = NULL) %>% collect(), gf)
+
+  # even if it doesn't exist
+  expect_dplyr_equal(input %>% mutate(z = NULL) %>% collect(), df)
+  # or was just created
+  expect_dplyr_equal(input %>% mutate(z = rep(1, nrow(input)), z = NULL) %>% collect(), df)
+
+  # regression test for https://github.com/tidyverse/dplyr/issues/4974
+  expect_dplyr_equal(
+    input %>% mutate(z = 1, x = NULL, y = NULL) %>% collect(),
+    data.frame(x = 1, y = 1)
+  )
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L71-L75
+test_that("assignments don't overwrite variables (dplyr #315)", {
+  expect_dplyr_equal(
+    input %>% mutate(z = {x <- 10; x}) %>% collect(),
+    tibble(x = 1, y = 2)
+  )
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L77-L81
+test_that("can mutate a data frame with zero columns and `NULL` column names", {
+  df <- vctrs::new_data_frame(n = 2L)
+  colnames(df) <- NULL
+  expect_dplyr_equal(
+    input %>% mutate(x = c(1,2)) %>% collect(),
+    df
+  )
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L102-L106
+test_that("mutate disambiguates NA and NaN (#1448)", {
+  expect_dplyr_equal(
+    input %>% mutate(y = x * 1) %>% select(y) %>% collect(),
+    tibble(x = c(1, NA, NaN))
+  )
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L117-L127
+test_that("mutate handles data frame columns", {
+  expect_dplyr_equal(
+    input %>% mutate(new_col = data.frame(x = 1:3)) %>% select(new_col) %>% collect(),
+    data.frame(x = 1:3)
+  )
+
+  # mutate() on grouped data not supported in Arrow; this will be pulling data back into R
+  expect_warning(expect_dplyr_equal(
+    input %>%
+      group_by(x) %>%
+      mutate(new_col = x) %>%
+      ungroup() %>%
+      select(new_col) %>%
+      collect(),
+    data.frame(x = 1:3)
+  ), "not supported in Arrow")
+
+  skip("rowwise() is not (yet) implemented in Arrow")
+  expect_dplyr_equal(
+    input %>%
+      rowwise(x) %>%
+      mutate(new_col = x) %>%
+      ungroup() %>%
+      select(new_col) %>%
+      collect(),
+    data.frame(x = 1:3)
+  )
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L129-L142
+test_that("unnamed data frames are automatically unspliced", {
+  # this works in arrow
+  expect_dplyr_equal(
+    input %>% mutate(tibble(b = 2)) %>% collect(),
+    tibble(a = 1)
+  )
+
+  # not this
+  expect_error(expect_dplyr_equal(
+    input %>% mutate(tibble(b = 2), tibble(b = 3)) %>% collect(),
+    tibble(a = 1)
+  ))
+
+  # nor this
+  expect_error(expect_dplyr_equal(
+    input %>% mutate(tibble(b = 2), c = b) %>% collect(),
+    tibble(a = 1)
+  ))
+})
+
+# similar to https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-mutate.r#L144-L148
+test_that("named data frames are packed", {
```

Comment:
I think this test ("named data frames are packed") should be removed; it's not testing anything about arrow besides that the data is pulled into an R data frame and the dplyr operation is performed on that